src/platform_linux/platform.c

```diff
@@ -19,6 +19,15 @@ bool platform_use_mlock   = FALSE;
 platform_log_handle *Platform_default_log_handle = NULL;
 platform_log_handle *Platform_error_log_handle   = NULL;
 
+/*
+ * Declare globals to track heap handle/ID that may have been created when
+ * using shared memory. We stash away these handles so that we can return the
+ * right handle via platform_get_heap_id() interface, in case shared segments
+ * are in use.
+ */
+platform_heap_handle Heap_handle = NULL;
+platform_heap_id     Heap_id     = NULL;
```

Comment:
`Heap_id` is needed to service the return from `platform_get_heap_id()`.

The sibling global is created for symmetry.